.github/workflows/android-build.yml

```
@@ -177,7 +177,11 @@ jobs:
       - name: Set MSBuild options
         # Configuire VS environment variables through the developer command prompt for MSVC
         uses: ilammy/msvc-dev-cmd@0b201ec74fa43914dc39ae48a89fd1d8cb592756 # v1.13.0
-                
+
+      - name: Setup gradle
+        uses: gradle/actions/setup-gradle@v4
```

For security reasons, use the commit hash instead of the version tag:
```suggestion
        uses: gradle/actions/setup-gradle@017a9effdb900e5b5b2fddfb590a105619dca3c3 #v4.4.2
```